### ISSUE TITLE
Knex version 0.20 and above conflicts with no-invalid-await

### ISSUE THREAD

**Issue by KristjanTammekivi** [SonarSource/SonarTS#889]
**I want to report a bug.**

**SonarTS version:** 1.9.0<br>
**Node.js version:** 12.13.0<br>
**TypeScript version:** 3.7.2<br>

**TSLint version:** 5.18.0<br>

**Rule key:** no-invalid-await

**Reproducer** 
```
interface SomeCallable {
    (cb: Function): Promise<any>;
}

interface Builder {
    then: SomeCallable;
}

async function testme(knex: Builder) {
    await knex; // Type of 'await' operand must either be a valid promise or must not contain a callable 'then' member 
}

```

**Expected behavior**

Knex typings were changed in 0.20.0 so they don't inherit from Promise anymore, but there are still .then/.catch methods and should not warn (initially reported this in https://github.com/knex/knex/issues/3516)
